tiled/_tests/adapters/test_arrow.py

```diff
@@ -0,0 +1,69 @@
+import tempfile
+import unittest
+
+import pyarrow as pa
+
+from tiled.adapters.arrow import ArrowAdapter
+from tiled.structures.table import TableStructure
+
+names = ["f0", "f1", "f2"]
+data0 = [
+    pa.array([1, 2, 3, 4, 5]),
+    pa.array(["foo0", "bar0", "baz0", None, "goo0"]),
+    pa.array([True, None, False, True, None]),
+]
+data1 = [
+    pa.array([6, 7, 8, 9, 10, 11, 12]),
+    pa.array(["foo1", "bar1", None, "baz1", "biz", None, "goo"]),
+    pa.array([None, True, True, False, False, None, True]),
+]
+data2 = [pa.array([13, 14]), pa.array(["foo2", "baz2"]), pa.array([False, None])]
+
+
+class TestArrowAdapter(unittest.TestCase):
```

Comment:
Since about 2016, scientific Python libraries have generally moved away from `unittest.TestCase` and preferred functional-style test construction. In the modern style, the setup functions becomes a fixture and the `test_` methods become functions, like this:

```py
@pytest.fixture
def adapter():
    # Do setup.
    return ArrowAdapter(...)

def test_attributes(adapter):
    assert adapter.structure().columns == names
    assert self.adapter.structure().npartitions == 3
```